### DISCUSSION TITLE
It should fail if it discover unrecognised property in any functions configuration

### DISCUSSION THREAD

**Discussion by daxaymakwana** [serverless/serverless#11257] [General]
Hi Guys,

I have one suggestion.

I accidentally write ```environnment``` instead of ```environment ```.

But serverless successfully completed the deployment with following warning:

```
2022-07-19T10:27:42.541-07:00	Serverless: at 'functions.my_api_tests': unrecognized property 'environnment'
```

I think it should fail.

Let me know your opinions on this.

**Comment by pgrzesik** [comment]
Hey @daxaymakwana - we already have that functionality, by default it produces a warning but you can configure it to an error by setting `configValidationMode: error` in your config. We considered switching to `error` by default, but some plugins do not define a schema which would cause more trouble to their users.
